quixstreams/sinks/influxdb_v3.py

```diff
@@ -0,0 +1,91 @@
+from itertools import islice
+from typing import Optional, Iterable, TypeVar
+
+import influxdb_client_3
+from influxdb_client_3 import InfluxDBClient3, WritePrecision
+
+from .base import Sink, SinkBatch
+from .exceptions import SinkBackpressureError
+
+T = TypeVar("T")
+
+
+class InfluxDBV3Sink(Sink):
+    def __init__(
+        self,
+        token: str,
+        host: str,
+        org: str,
+        database: str,
+        measurement: str,
+        fields_keys: Iterable[str] = (),
+        tags_keys: Iterable[str] = (),
+        time_key: Optional[str] = None,
+        batch_size: int = 1000,
+        debug: bool = False,
+    ):
+        super().__init__()
+        self._client = InfluxDBClient3(
+            token=token,
+            host=host,
+            org=org,
+            database=database,
+            debug=debug,
+        )
+        self._measurement = measurement
+        self._fields_keys = fields_keys
+        self._tags_keys = tags_keys or []
+        self._time_key = time_key
+        self._write_precision = WritePrecision.MS
+        self._batch_size = batch_size
+
+    def _iter_batches(self, it: Iterable[T], n: int) -> Iterable[Iterable[T]]:
+        """
+        Batch data into tuples of length n. The last batch may be shorter.
+        """
+        if n < 1:
+            raise ValueError("n must be at least one")
+        it_ = iter(it)
+        while batch := tuple(islice(it_, n)):
+            yield batch
+
+    def write(self, batch: SinkBatch):
+        measurement = self._measurement
+        fields_keys = self._fields_keys
+        tags_keys = self._tags_keys
+        time_key = self._time_key
+        for write_batch in self._iter_batches(batch, n=self._batch_size):
+            records = []
+            for item in write_batch:
+                value = item.value
+                tags = {tag_key: value[tag_key] for tag_key in tags_keys}
+                tags["__key"] = item.key
+                tags["__offset"] = item.offset
```

Comment:
This is definitely not a tag. It will very quickly cause high cardinality. Also although other tags looks handy I would consider if this is turned off as default. Not strong on that though...

Reply:
Okay, that makes sense. I probably misinterpreted the "tag" in the context of InfluxDB.

Reply:
Updated